### PR TITLE
feat: Add secure MessageBus capability

### PR DESCRIPTION
**Pull request by lenny-intel**
**The PR is dependent on edgexfoundry/go-mod-bootstrap#225**

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## What is the current behavior?
Secure MessageBus capability does not exists

## What is the new behavior?
Secure MessageBus capability now exists

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by hahattan — CHANGES_REQUESTED** [review]

File: internal/config/types.go [hunk 1]
```diff
@@ -56,6 +56,8 @@ type ServiceInfo struct {
 	AsyncBufferSize int
 	// MaxRequestSize defines the maximum size of http request body in bytes
 	MaxRequestSize int64
+	// UseMessageBus indicates whether or not the Event are published directly to the MessageBus
+	UseMessageBus bool
```
Comment (hahattan) [inline internal/config/types.go:60]: Remove `MessageQueueInfo` struct as it's now in go-mod-bootstrap (line107)

**Review by hahattan — CHANGES_REQUESTED** [review]

File: example/cmd/device-simple/res/configuration.toml [hunk 2]
```diff
@@ -41,11 +42,12 @@ Type = 'consul'
...
 
 [MessageQueue]
-Enabled = false
-Protocol = 'tcp'
+Protocol = 'redis'
 Host = 'localhost'
-Port = 1883
-Type = 'mqtt'
+Port = 6379
+Type = 'redisstreams'
```
Comment (cloudxxx8) [inline example/cmd/device-simple/res/configuration.toml:48]: are we still calling it 'redisstreams' when we are using redis pub/sub?  Are they the same?
I feel confused about the message type name.  How about just calling it 'redis'?

File: internal/clients/init.go [hunk 3]
```diff
@@ -56,10 +55,6 @@ func InitDependencyClients(ctx context.Context, wg *sync.WaitGroup, startupTimer
 	}
 	initCoreServiceClients(dic)
 
-	if configuration.MessageQueue.Enabled && initMessagingClient(ctx, wg, startupTimer, dic) == false {
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply internal/clients/init.go:59]: Good catch! THX.

File: internal/config/types.go (hunk 1 above)
Comment (lenny-intel) [reply internal/config/types.go:60]: Good catch! THX.

File: example/cmd/device-simple/res/configuration.toml (hunk 2 above)
Comment (lenny-intel) [reply example/cmd/device-simple/res/configuration.toml]: Done

**Comment by lenny-intel** [comment]
@cloudxxx8 , @hahattan , I have addressed for comments, ready for re-review.

**Review by hahattan — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**